### PR TITLE
Update docs

### LINKED ISSUE #2686: Documentation: ArduPilot support
Docs don't mention ArduPilot support.
https://microsoft.github.io/AirSim/search.html?q=ardupilot returns no results.
Linked issue: https://github.com/ArduPilot/ardupilot_wiki/issues/2739

### LINKED ISSUE #2745: Unable to view the Reinforcement Learning tutorial by Ashish Kapoor
Hello, 

I was trying to view the RL tutorial by Ashish Kapoor using the following link:
https://microsoft.github.io/AirSim/docs/reinforcement_learning

However, the page seems empty. Is this the intended behavior? 

Screenshot from 2020-06-03 18-54-28

### PR DESCRIPTION
**Pull request by rajat2004**
1. Add Python API docs
2. Add links for ArduPilot usage (Closes #2686)
3. Update CHANGELOG.md
4. Add more info about Logs in FAQs 
5. Fix indentation
6. Note about installation in C drive (#2697)
7. Fix RL page link (closes #2745)
8. Add  [PyMarkdown extras extension](https://facelessuser.github.io/pymdown-extensions/extensions/extra/) - requires `pip install pymdown-extensions`

I'm not quite sure as to how to link to the html page from the markdown files other than just having it in the nav bar and mentioning in the markdown to go to them through the github.io page nav bar. Better ways to add it to the docs are very much welcome

The 5th commit fixes the code-blocks and FAQs ordering in https://microsoft.github.io/AirSim/build_linux/, and numbering of steps in https://microsoft.github.io/AirSim/unreal_custenv/
It, however, causes the content of .uproject in the custom env page to be like the ones in the Linux build page. I thought it's okay considering it fixes the step numbering which I think is more imp
Let me know if I should change it back

Edit: The above is no longer a problem, the `extras` extension contains `superfences` which handles fenced codeblocks, and gives a nice-looking border around them, do test it out!

Just using `build_docs.sh` and opening the `index.html` file will lead to links opening to the file server and not the html files directly. Run `mkdocs serve` to see how it will actually work

### CODE REVIEW

**Comment by rajat2004** [comment]
Improvement would be to automate the docs build, that will prevent out-of-date docs at different places and save maintainer effort
Though that will most likely require an encrypted token key for say Travis to push to the gh-pages branch

**Comment by rajat2004** [comment]
For some reason, navigation using the sidebar is having the same behaviour for all the links, not just the API docs. It's also happening for me on the master branch.
Any idea about this?

**Comment by rajat2004** [comment]
@madratman @saihv Could you have a look at this? Main addition here required is to enable automatic docs build using say Travis or Github Actions. I'm not sure as to what method is used currently, is it run `build_docs.sh` from master branch, then `mkdocs gh-deploy`?
Note that this will most likely involve creating a new token with push access to the Airsim repository, and adding that to Travis

**Comment by rajat2004** [comment]
Rebased on master

**Comment by rajat2004** [comment]
Added a commit which runs a currently non-existent `deploy_docs.sh` script on Travis master
I'm not sure of how the docs are updated, but probably `mkdocs gh-deploy` isn't used since it erases all history on the branch.

Maybe something like https://github.com/mkdocs/mkdocs/pull/1082? I went for Travis instead of a Github Action since we also Sphinx in the mix to build the Python API docs

**Comment by rajat2004** [comment]
This will conflict with https://github.com/microsoft/AirSim/pull/2712, will be better to merge that first
Also, the last commit could be separated out into a different PR since setting up auto deployment will probably take some time, if it's preferable to get the other docs updates in first

**Comment by rajat2004** [comment]
Moved Travis commits to #2749

**Comment by rajat2004** [comment]
Ping!

**Comment by rajat2004** [comment]
@madratman @saihv Is it possible to get this one merged anytime soon? The API reference would be very helpful for figuring out problems when using the Python APIs, and not require a deep-dive into the code to figure things out

File: build_docs.sh [hunk 1]
```diff
@@ -1,5 +1,11 @@
...
 
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+pushd "$SCRIPT_DIR"  >/dev/null
+
 cp README.md docs/
 sed -i 's/](docs\//](/g' docs/README.md
+
 mkdocs build
+
+popd >/dev/null
```
[the remark this answers is not in the document]
Comment (madratman) [reply build_docs.sh:11]: I'll copy generated api docs to a dir build_docs/ (not docs/)

**Pull request merged**